che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -418,8 +419,24 @@ public void clone(CloneRequest request) throws GitException, UnauthorizedExcepti
                 deleteRepositoryFolder();
             }
 
-            throw new GitException(exception.getMessage());
+            String message = generateSSLExceptionMessage(exception);
+            throw new GitException(message, exception);
+        }
+    }
+
+    private String generateSSLExceptionMessage(Throwable e) {
```

Comment:
I think the name of the method is not clear enough. It says that method should give 'SSLException' message but given exception can be an in instance of any exception type, not only 'SSLException'.